browser/lib/util/webstorage.js

```diff
@@ -9,25 +10,26 @@ var WebStorage = (function() {
 	 * somewhat roundabout way. (If unsupported or no window object,
 	 * will throw on accessing a property of undefined) */
 	try {
-		window.sessionStorage.setItem(test, test);
-		window.sessionStorage.removeItem(test);
+		globalObject.sessionStorage.setItem(test, test);
+        globalObject.sessionStorage.removeItem(test);
 		sessionSupported = true;
 	} catch(e) {
 		sessionSupported = false;
 	}
 
 	try {
-		window.localStorage.setItem(test, test);
-		window.localStorage.removeItem(test);
+        globalObject.localStorage.setItem(test, test);
+        globalObject.localStorage.removeItem(test);
 		localSupported = true;
 	} catch(e) {
 		localSupported = false;
 	}
 
-	function WebStorage() {}
+
+    function WebStorage() {}
 
 	function storageInterface(session) {
-		return session ? window.sessionStorage : window.localStorage;
+		return session ? globalObject.sessionStorage : globalObject.localStorage;
```

Comment:
> a webworker also has no window variable; but does have a global object with webstorage attached

Are you sure? I've never used webworkers, but from what I can tell from a quick google (eg [this](http://stackoverflow.com/a/6179599)), that doesn't seem to be correct?

> doesn't ably by default try WS then XHR

No, it starts with the lowest-common-denominator thing (xhr) then upgrades to ws where it can. That avoids having a delay before connection on old platforms. Then once it does upgrade, it stores websocket as a preference, so next time it knows it can go straight for that. That's why I'm keen to have localstorage if possible.

Any thoughts on my earlier suggestion:

> If that's the case, then perhaps we should have `sessionsSupported` be false, but `localSupported` be true and `WebStorage.get/set/remove` implemented here as a layer around the nativescript `application-settings` mechanism. What do you think?

> For cleanness, if we're doing that, then should probably implement a separate `webstorage.js` for nativescript and include that separately in the grunt concatenator (ie split it out of `ablyFiles`).